pkg/main.go

```diff
@@ -25,7 +25,7 @@ func main() {
 	// for an InstanceFactoryFunc in grafana-plugin-sdk-go
 	// TODO: Remove this function and create a NewDatasource method with Context in grafana-aws-sdk, see https://github.com/grafana/oss-plugin-partnerships/issues/648
 	newDatasourceForUpgradedPluginSdk := func(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
-		return ds.NewDatasource(settings)
+		return ds.NewDatasource(ctx, settings)
```

Comment:
Can we remove `newDatasourceForUpgradedPluginSdk` now that `ds.NewDatasource` takes a context argument?

Also in a previous PR from where the TODO comment above came from, a `Context.TODO()` was added to api.go as well https://github.com/grafana/athena-datasource/pull/290/files#diff-92b959f50b68ee1be332080f2c7213c4df62268a536b60dc10bd1ca382030930. Can that also use the context now as well or is there anything else that it's waiting for?

Reply:
Great catch! I fixed this one since it was easypeasy! That API one is used in a few places and was ever so slightly less straightforward so I'd like to save that for a future pr someday, but I'll move that ticket into the bottom of next!